Circuit/SymbolLayout.cs

```diff
@@ -99,6 +103,12 @@ public void AddCurve(EdgeType Type, IEnumerable<Coord> Points)
 
             DrawCurve(Type, Points.Select(i => (Point)i));
         }
+
+        internal void DrawArc(EdgeType type, Coord center, double radius, double startAngle, double endAngle, Direction direction = Direction.Clockwise)
```

Comment:
Rather than add a new type of shape, I wonder if all circles should just have startAngle and endAngle, defaulted to 0 and 2pi. I also wonder if you need the `Direction`, can't you just use `startAngle, endAngle = 0, pi` and `pi, 2pi` for the arcs in your new symbol?